Source/Plugin.BLE.iOS/Adapter.cs

```diff
@@ -420,6 +420,19 @@ private async Task<CBPeripheral> ScanForPeripheralAsync(Func<IDevice, bool> devi
                 linkedToken.Register(() => taskCompletionSource.TrySetCanceled());
 
                 DeviceDiscovered += handler;
+
+                // We could already be scanning, if that's the case, check if we have found any devices that matches
+                if (IsScanning)
+                {
+                    var foundDevice = DiscoveredDevices.FirstOrDefault(x => deviceFilter(x));
```

Comment:
I'm unsure if this is a good idea. Could the DiscoveredDevices contain products that we cannot find anymore?

Reply:
```if (foundDevice?.NativeDevice is CBPeripheral nativeDevice)
{
  return nativeDevice;
}
```

nicer? nah maybe your approach is the one to go with